### PR TITLE
Ruby productions for identifier and call nodes

### PR DESCRIPTION
**Pull request by rahul-privado**
### Description
1. Implemented most productions so as to get all identifier and non-operator call nodes
2. Unit test changes to account for this

### Testing done
1. Unit tests have been modified to account for these changes. Each unit test code has been put in the ANTLR lab to view the parse tree and verify that all identifiers and non-operator call nodes get into the AST
2. Imported 3 Ruby repositories downloaded from the internet and made sure that they get imported. Fixed exceptions that showed up.

The unit tests and importing repositories resulted in execution of most of the AST flows. Several exceptions were encountered and fixed. 

### Please note
Please do not remove the `case _ => Ast()` blocks with a `All contexts mismatched` log from the code. There are parsing issues with the parser and so those blocks do get hit with the downloaded repositories since no production matches.` case _ => Ast()`  be removed once the parser is completely fixed.

### CODE REVIEW

File: joern-cli/frontends/rubysrc2cpg/src/main/antlr4/io/joern/rubysrc2cpg/parser/RubyParser.g4 [hunk 1]
```diff
@@ -110,16 +110,16 @@ primary
 // --------------------------------------------------------
 
 singleLeftHandSide
-    :   variableIdentifier
-    |   primary LBRACK arguments? RBRACK
-    |   primary (DOT | COLON2) (LOCAL_VARIABLE_IDENTIFIER | CONSTANT_IDENTIFIER)
-    |   COLON2 CONSTANT_IDENTIFIER
+    :   variableIdentifier                                                                                          # variableIdentifierOnly
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/main/antlr4/io/joern/rubysrc2cpg/parser/RubyParser.g4:113]: W.r.t. to labels, please follow the convention of suffixing the rule (name) they are part of. To wit, lables under `expressionOrCommand` are named `invocationExpressionOrCommand`, `notExpressionOrCommand`, etc. This shall avoid any future name collisions and hopefully be easier to reason about.

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala [hunk 2]
```diff
@@ -54,16 +54,24 @@ class IdentifierTests extends RubyCode2CpgFixture {
         |  end
         |end
         |
+        |Person p
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala:57]: `Person p` doesn't mean what you probably intended it to (and isn't meaningful Ruby): here, it's being parsed as a command (i.e. the invocation of a method named `Person` with argument `p`.)

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala [hunk 3]
```diff
@@ -165,8 +177,73 @@ class IdentifierTests extends RubyCode2CpgFixture {
         cpg.identifier.name("endvar").l.size shouldBe 2
         cpg.identifier.name("beginbool").l.size shouldBe 1
         cpg.identifier.name("endbool").l.size shouldBe 1
+        cpg.call.name("puts").size shouldBe 1
         cpg.identifier.size shouldBe 6
       }
     }
+
+    "The CPG generated for do block" should {
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala:185]: Suggestion: I think this test title could be confusing, as it's only looking for identifiers. Proper do-block tests should probably reside in a different test class.

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala [hunk 4]
```diff
@@ -165,8 +177,73 @@ class IdentifierTests extends RubyCode2CpgFixture {
...
+      val cpg = code(
+        """
+          |[1, 2, "three"].each do |n|
+          | puts n
+          |end
+          |""".stripMargin,
+        fileName = "doblock.rb"
+      )
+
+      "doblocktest" in {
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala:195]: A more suggestive name would be ideal. (Here and below)

**Review by xavierpinho — CHANGES_REQUESTED** [review]
I'd just kindly ask for a clean-up round on labels and test names before merging.

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala (hunk 3 above)
Comment (rahul-privado) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala]: Changed the title. Please check

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala (hunk 4 above)
Comment (rahul-privado) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala]: Changed the name

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala (hunk 2 above)
Comment (rahul-privado) [reply joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/IdentifierTests.scala]: Fixed this.

File: joern-cli/frontends/rubysrc2cpg/src/main/antlr4/io/joern/rubysrc2cpg/parser/RubyParser.g4 (hunk 1 above)
Comment (rahul-privado) [reply joern-cli/frontends/rubysrc2cpg/src/main/antlr4/io/joern/rubysrc2cpg/parser/RubyParser.g4]: Agreed and done. Some of the names are really long however. Please let me know if you have any further suggestions. Marking this resolved.

**Review by xavierpinho — APPROVED** [review]

**Pull request merged**